src/parser_urdf.cc

```diff
@@ -2793,6 +2833,45 @@ void CreateLink(tinyxml2::XMLElement *_root,
   _root->LinkEndChild(elem);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+void CreateFrameFromLink(tinyxml2::XMLElement *_root,
+                         urdf::LinkConstSharedPtr _link,
+                         const gz::math::Pose3d &_currentTransform)
+{
+  // create new body
+  tinyxml2::XMLElement *elem = _root->GetDocument()->NewElement("frame");
+
+  // set body name
+  elem->SetAttribute("name", _link->name.c_str());
+
+  // compute global transform
+  gz::math::Pose3d localTransform;
+  // this is the transform from parent link to current _link
+  // this transform does not exist for the root link
+  if (_link->parent_joint)
+  {
+    tinyxml2::XMLElement *pose = _root->GetDocument()->NewElement("pose");
+    pose->SetAttribute("relative_to", _link->parent_joint->name.c_str());
```

Comment:
I think we also need to set the `attached_to` attribute for the frame. Otherwise, it will be attached to the canonical link, which we don't want. For that to work though, I believe either the parent joint or the child joint has to be a "fixed" joint. We would then attach the frame to the joint that has a fixed type. Otherwise, we would need to emit an error and not create the frame.

Reply:
Added `attached_to`, which targets either a parent of child joint frame that is a fixed joint, https://github.com/gazebosim/sdformat/pull/1238/commits/1e5a2da98984178ccf9812b0cc3c0c181c456ecf